### PR TITLE
Added recipe for mongomock

### PR DESCRIPTION
**Pull request by pmlandwehr**
I'm pulling the source from github because [pypi](https://pypi.python.org/pypi/mongomock/3.6.0) only has a wheel.

### CODE REVIEW

File: recipes/mongomock/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{%set name = "mongomock" %}
+{%set version = "3.6.0" %}
+{%set compress_type = "tar.gz" %}
```
Comment (patricksnape) [inline recipes/mongomock/meta.yaml]: This isn't used?

**Comment by pmlandwehr** [comment]
Restarting...

File: recipes/mongomock/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+{%set version = "3.6.0" %}
+{%set build_num = "0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/mongomock/{{ name }}.git
+  git_rev: {{ version }}
```
Comment (jakirkham) [inline recipes/mongomock/meta.yaml]: Can we please change this to a tarball and have a checksum?

e.g. https://github.com/mongomock/mongomock/archive/{{ version }}.tar.gz

**Comment by jakirkham** [comment]
Thoughts on this @pmlandwehr ?

File: recipes/mongomock/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+{%set hash_type = "sha256" %}
+{%set hash_val = "2e6b85f97b84d1797c5b0deceff72690a2e79f82c8ec410663a4175a83b8a80f" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ version }}.{{ compress_type }}
+  url: https://github.com/mongomock/mongomock/archive/{{ version }}.{{ compress_type }}
```
Comment (jakirkham) [inline recipes/mongomock/meta.yaml]: Did you try PyPI? Or did that not work? If so, can you please point me to a sample build?

Comment (pmlandwehr) [inline recipes/mongomock/meta.yaml]: [PyPI](https://pypi.org/project/mongomock/#files) only has a wheel, unfortunately. This is definitely a bit odd -  there's certainly an sdist at [the url].(https://github.com/mongomock/mongomock/archive/3.6.0.tar.gz). I'm kind of flummoxed. Maybe an issue with `pbr`? ([reference](https://github.com/ekarlso/python-odlclient/issues/3))

**Pull request closed without merging**